cpp/src/fil/infer.cu

```diff
@@ -136,13 +131,30 @@ See https://rapids.ai/start.html as well as cmake defaults.
 */
 // values below are defaults as of this change.
 template <int NITEMS>
-using BlockReduceHost =
-  typename cub::BlockReduce<vec<NITEMS, float>, FIL_TPB,
-                            cub::BLOCK_REDUCE_WARP_REDUCTIONS, 1, 1, 600>;
+size_t block_reduce_footprint_host() {
```

Comment:
I was changing half the cases where this was used, as well as deleting the device-side `template using` statements. This meant the host ones were left alone and did not need to define class separately from footprint. This allowed to implement smem footprint additions for GROVE_PER_CLASS_* in a much more readable way (see below).